src/diffusers/configuration_utils.py

```diff
@@ -567,7 +590,11 @@ def to_json_string(self) -> str:
                 String containing all the attributes that make up the configuration instance in JSON format.
         """
         config_dict = self._internal_dict if hasattr(self, "_internal_dict") else {}
-        config_dict["_class_name"] = self.__class__.__name__
+        cls_name = self.__class__.__name__
+        if cls_name not in _all_available_pipeline_component_modules:
+            config_dict["_class_name"] = [str(self.__class__.__module__), cls_name]
```

Comment:
So that `model_index.json` can have the `_class_name` like so:
https://huggingface.co/hf-internal-testing/tiny-sdxl-custom-all/blob/main/model_index.json#L2

Otherwise, there's no way to achieve this I think.

Reply:
@sayakpaul It is best to still support the original version and call save_pretrained and from_pretrained from the custom pipeline class to work normally, if there are no disadvantages. Personally, I don't feel the need to distinguish between custom and diffusers. I didn't care about this during my original use and it worked well.